src/utils/account-with-lockup.js

```diff
@@ -34,47 +34,53 @@ async function signAndSendTransaction(receiverId, actions) {
     const lockupAccountId = getLockupAccountId(this.accountId)
     if (missingAmount.gt(new BN(0)) && (await accountExists(this.connection, lockupAccountId))) {
         console.warn('Not enough balance on main account, checking lockup account', lockupAccountId);
-
         if (!(await this.wrappedAccount.viewFunction(lockupAccountId, 'are_transfers_enabled'))) {
             await this.wrappedAccount.functionCall(lockupAccountId, 'check_transfers_vote', {}, BASE_GAS.mul(new BN(3)))
         }
-
-        const lockedBalance = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_locked_amount'))
-        const poolAccountId = await this.wrappedAccount.viewFunction(lockupAccountId, 'get_staking_pool_account_id')
-        if (!poolAccountId && lockedBalance.eq(new BN(0))) {
-            console.info('Destroying lockup account to claim remaining funds', lockupAccountId)
-
-            const newKeyPair = KeyPair.fromRandom('ed25519')
-            await this.wrappedAccount.functionCall(lockupAccountId, 'add_full_access_key', {
-                new_public_key: newKeyPair.publicKey.toString()
-            }, BASE_GAS.mul(new BN(2)))
-
-            const tmpKeyStore = new InMemoryKeyStore()
-            await tmpKeyStore.setKey(this.connection.networkId, lockupAccountId, newKeyPair)
-            const tmpConnection = new Connection(this.connection.networkId, this.connection.provider, new InMemorySigner(tmpKeyStore))
-            const lockupAccount = new Account(tmpConnection, lockupAccountId)
-            await lockupAccount.deleteAccount(this.accountId)
-        } else {
-            let liquidBalance = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_liquid_owners_balance'))
-            if (!liquidBalance.gt(missingAmount)) {
-                await this.wrappedAccount.functionCall(lockupAccountId, 'refresh_staking_pool_balance', {}, BASE_GAS.mul(new BN(3)))
-                liquidBalance = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_liquid_owners_balance'))
-                if (!liquidBalance.gt(missingAmount)) {
-                    throw new WalletError('Not enough tokens.', 'signAndSendTransactions.notEnoughTokens')
-                }
-            }
-
-            await this.wrappedAccount.functionCall(lockupAccountId, 'transfer', {
-                // NOTE: Move all the liquid tokens to minimize transactions in the long run
-                amount: liquidBalance.toString(),
-                receiver_id: this.wrappedAccount.accountId
-            }, BASE_GAS.mul(new BN(2)))
-        }
+    
+        await transferAllFromLockup(lockupAccountId, missingAmount)
```

Comment:
@Patrick1904 needs to use `this.`, i.e. `await this.transferAllFromLockup(lockupAccountId, missingAmount)`